docs/standard/class-libraries-multi-targeting.md

```diff
@@ -0,0 +1,339 @@
+the consuming project.
+
+To produce a NuGet package you need to configure your project accordingly. You
+can do this in Visual Studio by right clicking the project and selecting
+**Properties**. On the **Package** tab you need to check the box labelled
+**Generate NuGet package on build**.
+
+You can also enable package generation by editing the project file:
+
+```xml
+<Project Sdk="Microsoft.NET.Sdk">
+
+  <PropertyGroup>
+    <GeneratePackageOnBuild>true</GeneratePackageOnBuild>
+  </PropertyGroup>
+
+</Project>
+```
+
+# Best practices
+
+When doing multiple targeting it's important to follow a set of rules in order
+to ensure you and the consumers of your code have a great experience:
+
+* **DO** start with .NET Standard 2.0. Most general purpose libraries will not
+  need APIs outside this set.
+
+* **CONSIDER** adding framework specific implementations if you need to call
+  APIs outside the .NET Standard API set.
+
+* **CONSIDER** using multi-targeting over multiple projects when you need to add
+  platform specific outputs in libraries.
+
+* **CONSIDER** using the community package `MSBuild.Sdk.Extras` when using
+  multi-targeting. It adds a bunch of properties and targets that don't exist in
+  MSBuild and keeps your project files lean & clean.
+
+* **CONSIDER** using multiple projects over multi-targeting when you need to add
+  platform specific outputs for applications.
+
+* **DO NOT** drop support for .NET Standard when producing libraries for
+  multiple frameworks. Instead, throw from the implementation and offer
+  capability APIs. This shields your consumers from having to target multiple
+  frameworks as well. And should you add support for new frameworks later on,
+  your consumers run in more places without having to change their code.
+
+* **DO** throw `PlatformNotSupportedException` when you cannot provide a
+  meaningful implementation for .NET Standard.
+
+* **DO** provide capability APIs for features that can throw
+  `PlatformNotSupportedException`. This allows consumers to use optional
+  features without having to catch exceptions.
+
+* **DO** share your component using a single NuGet package. It shields consumers
+  from having to choose the appropriate binary.
+
+## Not dropping support for .NET Standard
+
+As a library author your goal is generally to encapsulate a piece of
+functionality. As mentioned earlier, when doing multi-targeting you don't want
+your customers having to understand which binary to select, which is why
+recommend that you generate a NuGet package.
+
+This also means you generally want to produce a .NET Standard output when you
+use multi-targeting. This allows your consumers to be .NET Standard without
+having to use multi-targeting. In other words, from your customer's point of
+view they don't even have to be aware that you used multi-targeting -- it's
+an implementation detail of your library.
+
+This practice is often called *bait & switch*:
```

Comment:
Can we pick a different name? "bait & switch" has negative connotations. I think you mean this to be a positive where the details of how to get multiple platforms are hidden.